### PR TITLE
Add claim_establishments foreign key to dispatch_tasks

### PR DESCRIPTION
**Pull request by yoomlam**
### Description
This is phase 13 of a multi-phase plan to add missing foreign keys: `claim_establishments` to `dispatch_tasks` via `task_id`.

Phase 12 PR #16922
Phase 11 PR #16556 
Phase 10 PR #16513
Phase 9 PR #16366
Phase 8 PR #16327
Phase 7c PR #16313
Phase 7b PR #16304
Phase 7 PR #16277
Phase 5c PR #16263
Phase 5b PR #16281
Phase 4 PR #16137
Phase 3 PR #16120
Phase 2 PR #16114
Phase 1 PR #16034 has more details

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
```ruby
ces=ClaimEstablishment.unscoped.includes(:task).where.not(task_id: nil).select{|r| r.task == nil};
ces.count
=> 0 #Prod
=> 0 #Preprod
=> 0  #UAT
```

- [Prod data fixed](https://dsva.slack.com/archives/CQTDX9BF0/p1636044142047700?thread_ts=1635868050.031300&cid=CQTDX9BF0)
- [Preprod data fixed](https://dsva.slack.com/archives/CQTDX9BF0/p1637011383009300?thread_ts=1637009871.007800&cid=CQTDX9BF0)
- [UAT data fixed](https://dsva.slack.com/archives/CQTDX9BF0/p1637078112010100?thread_ts=1637009871.007800&cid=CQTDX9BF0)

### Database Changes

* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [x] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [x] Post this PR in #appeals-schema with a summary

### CODE REVIEW

File: config/initializers/immigrant.rb [hunk 1]
```diff
@@ -42,5 +39,6 @@
 
   # Don't need FKs on a cache table
   { from_table: "cached_appeal_attributes", column: "appeal_id" },
-  { from_table: "cached_appeal_attributes", column: "vacols_id" }
+  { from_table: "cached_appeal_attributes", column: "vacols_id" },
+  { from_table: "cached_user_attributes", column: "sdomainid" }
```
Comment (yoomlam) [inline config/initializers/immigrant.rb:43]: Added line 43 to ignore suggestion from running `make check-fks`:
```
Missing foreign key relationship on 'cached_user_attributes.sdomainid' to 'users.css_id'
Found missing foreign keys, run `rails generate immigration MigrationName` to create a migration to add them.
```

File: config/initializers/immigrant.rb [hunk 2]
```diff
@@ -7,9 +7,6 @@
   # See https://dsva.slack.com/archives/C3EAF3Q15/p1636061857138600?thread_ts=1636060646.138000&cid=C3EAF3Q15
   { from_table: "available_hearing_locations", column: "veteran_file_number" },
 
-  # Add FK to dispatch_tasks table (not the tasks table)
-  { from_table: "claim_establishments", column: "task_id" },
-
```
Comment (yoomlam) [inline config/initializers/immigrant.rb:12]: We're adding the FK in this PR, so we no longer need to ignore it.

File: db/migrate/20211118214040_validate_foreign_key_for_claim_establishment_task.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class ValidateForeignKeyForClaimEstablishmentTask < ActiveRecord::Migration[5.2]
```
Comment (n1zyy) [inline db/migrate/20211118214040_validate_foreign_key_for_claim_establishment_task.rb:1]: Does this not need to inherit from `Caseflow::Migration`? It seems like it probably doesn't, but want to make sure.

**Review by n1zyy — APPROVED** [review]
1 question, but otherwise looks like a good improvement. 👍

File: db/migrate/20211118214040_validate_foreign_key_for_claim_establishment_task.rb (hunk 3 above)
Comment (yoomlam) [reply db/migrate/20211118214040_validate_foreign_key_for_claim_establishment_task.rb:1]: I'll update that.

**Pull request merged**